src/rand.rs

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018 Adgear
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+//! Random number generator based of the PCG paper (http://www.pcg-random.org/paper.html).
+
+use std::u32;
+use std::time::*;
+
+pub struct Rng {
+    state: u64,
+    inc: u64,
+}
+
+impl Default for Rng {
+    fn default() -> Self {
+        match SystemTime::now().duration_since(UNIX_EPOCH) {
+            Ok(res) => Self::seed_with(res.as_secs()),
```

Comment:
I've been thinking about this, but I wonder if providing a contract with respect to object initialization, that a seed must be provided by the user of the library.

Security weary people can worry about this in their application, and implement something more reasonable, rather than trying to add more complexity in mini.

Users that need a simpler rand module, and don't care about security can quickly dish out some code. 

The last case I can think of is of people that are unaware of these attacks, but I'm not sure what the best pedagogical approach is in this case.

Reply:
To seed an PRNG you're better off the micro/nano seconds then seconds. You can also use the `rdtsc` cpu instruction if you want to get real fancy.

Otherwise, I half agree with your points @psyomn but I don't think there's value in providing a sane default seed to reduce the surface for bugs.

Reply:
@RAttab I'm not sure what's your opinion. So, do we use the suggestion of @gnuvince?

Reply:
I think he means go with @gnuvince 's suggestion but use micro/nano seconds or rdtsc (if that's easy in Rust I guess).

Reply:
@antoyo Yep, micros and nanos for the seed is the way to go. I have a feeling that accessing rdtsc from rust will be painful so let's not bother with that one.